### PR TITLE
fix(node): remove deprecation warnings

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/22116

### CODE REVIEW

**Review by mmastrac — APPROVED** [review]
LGTM

File: ext/node/polyfills/internal_binding/stream_wrap.ts [hunk 1]
```diff
@@ -46,6 +46,8 @@ import {
 } from "ext:deno_node/internal_binding/async_wrap.ts";
 import { codeMap } from "ext:deno_node/internal_binding/uv.ts";
 
+const DENO_RID_SYMBOL = Symbol.for("Deno.internal.rid");
```
Comment (iuioiua) [inline ext/node/polyfills/internal_binding/stream_wrap.ts:49]:
```suggestion
const DENO_RID_SYMBOL = SymbolFor("Deno.internal.rid");
```
Nit: primordials.

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Comment by irbull** [comment]
I don't get it, although maybe take my opinion lightly.
1. If `rid` is really depreciated and going away, doesn't this "fix" really just hide it now? If / when `rid` is removed, this will now just fail at runtime.
2. There were several deprecations listed in 1.40. This fix only seems to address the one (rid). Maybe this is the only one that the node compat layer uses.

**Comment by bartlomieju** [comment]
> I don't get it, although maybe take my opinion lightly.
> 
> 1. If `rid` is really depreciated and going away, doesn't this "fix" really just hide it now? If / when `rid` is removed, this will now just fail at runtime.

Resource IDs are here to stay for some foreseeable future, but we want to make them an internal notion, so that we can migrate APIs to other mechanisms (like GCable objects like in https://github.com/denoland/deno/pull/21999). This is a stop-gap solution to stop the warnings. I'm working on a better solution that will use a unique symbol to store resource IDs (this symbol will not be reachable by user code so you won't have a way to access the value of the field).

> 2. There were several deprecations listed in 1.40. This fix only seems to address the one (rid). Maybe this is the only one that the node compat layer uses.

These appear to be the only ones that we missed :)

**Pull request merged**